src/domain/registration/form/RegistrationForm.tsx

```diff
@@ -33,14 +30,49 @@ import { profileQuery as ProfileQueryType } from '../../api/generatedTypes/profi
 import LoadingSpinner from '../../../common/components/spinner/LoadingSpinner';
 import NavigationConfirm from '../../../common/components/confirm/NavigationConfirm';
 import { GuardianInput, Language } from '../../api/generatedTypes/globalTypes';
-import { validateEmail } from '../../../common/components/form/validationUtils';
+import FormikDropdown, {
+  HdsOptionType,
+} from '../../../common/components/form/fields/dropdown/FormikDropdown';
+import { RegistrationFormValues } from '../types/RegistrationTypes';
+
+const schema = yup.object().shape({
+  guardian: yup.object().shape({
+    firstName: yup
+      .string()
+      .required('validation.general.required')
+      .max(255, 'validation.maxLength'),
+    lastName: yup
+      .string()
+      .required('validation.general.required')
+      .max(255, 'validation.maxLength'),
+    email: yup
+      .string()
+      .email('registration.form.guardian.email.input.error')
+      .required('validation.general.required'),
+    phoneNumber: yup
+      .string()
+      .required('validation.general.required')
+      .max(255, 'validation.maxLength'),
+    language: yup.string().max(255, 'validation.maxLength'),
+  }),
+  children: yup.array().of(
+    yup.object().shape({
+      postalCode: yup.string().required('validation.general.required'),
+      // relationship: yup.object().shape({
+      //   type: yup.string().required('validation.general.required').nullable(),
+      // }),
+    })
+  ),
+  agree: yup.boolean().oneOf([true], 'validation.required'),
+});
 
 const RegistrationForm: FunctionComponent = () => {
   const { i18n, t } = useTranslation();
   const currentLocale = getCurrentLanguage(i18n);
   const history = useHistory();
   const dispatch = useDispatch();
   const [isOpen, setIsOpen] = useState(false);
+  const [checked, setChecked] = useState(false);
```

Comment:
Seems like you are planning to have this value in the validation schema which alludes that you are planning to have the value in formik's state. In that case you can use formik's state to store the value and don't need this state. Note that you likely want to clean this field out before forwarding the values.

You can use Formik and yup in a few other ways to handle a value you don't want to forward, but want to validate. Cleaning it out before sending is probably the easiest and simplest one.

Reply:
I refactored this to using a special Terms checkbox component, based on your implementation in another project, pretty clever and nice solution.